coremltools/converters/mil/backend/nn/op_mapping.py

```diff
@@ -1375,12 +1375,15 @@ def lstm(const_context, builder, op):
         _squeeze(builder, op.outputs[2].name, output_names[2], axes=[0, 3, 4])
 
     elif direction == "bidirectional":
+        n_layers = len(builder.nn_spec.layers)
+        postfix_h = "_expanded_" + str(n_layers)
+        postfix_c = "_expanded2_" + str(n_layers)
         # Expand initial_h and initial_c
-        _expand_dim(builder, initial_h + "_expanded", initial_h, [2, 3, 4])
-        initial_h += "_expanded"
+        _expand_dim(builder, initial_h + postfix_h, initial_h, [2, 3, 4])
```

Comment:
Looks like this is due to `initial_h` and c being shared across multiple LSTM layers.
With this we will end up adding multiple expand dims. 
Another option is to have `set` maintained and skip expand if if already present but it's good not to pollute conversion with such a book-keeping and instead we can have a graph pass to remove such expand dims and other nodes

Reply:
I've gotten this error before in conv which I've solved with similar patch as you. I think we should do a global set in op_mapping.py, or this will keep haunting us

Reply:
Thank you for your comments :) You said:
>Looks like this is due to initial_h and c being shared across multiple LSTM layers.

I guess I need a clarification please. When we have 2 layers of LSTMs, we will have 2 different  h_0/c_0 with potentially different shapes as they are layer dependent. Therefore, I thought  "initial_h" and "initial_c" are layer dependent (due to potential different shapes). 

Your next comment says
>Another option is to have set maintained and skip expand if if already present
but that would only make sense if there is indeed one global initial_h (for the first layer only?) right? I mean, having the same name "initial_h" in 2 or N different layers for tensors of different shapes make little sense I think?

So, could you clarify please whether you expect 
a) initial_h and initial_c only apply to the 1st layer and therefore are only created once?
b) they are layer dependent with different shapes and therefore we need my patch, or something similar.

If (b) then my patch would be ok. 
if (a) then we can look at a different patch I think, like ask the builder whether the name already exists or something like that. @daiwei89 I am not sure how that would work. Wanna pitch in :) ?

Reply:
PS: for (a), if we go that way, which we need to know (hint, waiting feedback ;), then checking like 
"if (initial_h + "_expanded") in set(builder.layers):"
would do the trick, right?

Reply:
This will be seen in cases when we have more than one lstm layers and all have initial_h as `0` or some other same value and possibly MIL optimization passes optimized them away to single const. Let's consider case when we expect LSTM to have different `initial_h` in which case they already have different input name and ideally should not collide. 

I think what @daiwei89 is referring to is to maintain global set e.g. `seen_expand_dims` where we add entry whenever we add expand dim for any node and we add `expand_dim` only if it's not present for current input.

```
if initial_h not in seen_expand_dims:
    expand_dim(builder, initial_h + postfix_h, initial_h, [2, 3, 4])
    seen_expand_dims.insert(initial_h)
```

@daiwei89 note that, with this we need to reset the set before conversion starts otherwise if someone imports coremltools and converts more than one model in same context previous set is going to interfere into incorrect conversion for subsequent conversions.

Additionally, we can also log axes to ensure we can throw error if we find mismatch!

Reply:
Thanks for the clarification. So we do (a).
While I can contribute a few lines to do the local set check, like "if (initial_h + "_expanded") in set(builder.layers):", going the global set route is a bit more time-consuming I think. Therefore, is there value in me making a quick "local set" fix tomorrow, or should I hold off and leave it for someone else to make the full solution ?

Reply:
Changed the PT to reflect our discussion. Purely local solution for BLSTM. 
While I have a look at the test case, is this better?